bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/internal/firmware/FirmwareUpdateServiceTest.java

```diff
@@ -831,13 +920,14 @@ private void updateInvalidConfigAndAssert(int period, int delay, TimeUnit timeUn
 
     private void assertResultInfoEvent(ThingUID thingUID, Firmware firmware, String messageKey, Locale locale,
             String text, int expectedEventCount) {
-        firmwareUpdateService.removeFirmwareUpdateHandler(handler1);
-        firmwareUpdateService.addFirmwareUpdateHandler(handler1);
+        // FIXME See why remove and add
```

Comment:
I guess this should be checked

Reply:
I have rechecked that and it seems to be there for legacy reasons. The tests are behaving well without these statements so I will delete them.